libs/perun/services/src/lib/init-auth.service.ts

```diff
@@ -109,6 +110,10 @@ export class InitAuthService {
       if (perunPrincipal.user === null) {
         const config = getDefaultDialogConfig();
         this.dialog.open(UserDontExistDialogComponent, config);
+      } else if (perunPrincipal.user.majorSpecificType === 'SERVICE') {
+        this.authService.logout();
```

Comment:
Depending on the proxy and configuration the logout might redirect you without the dialog ever showing. But even if the logout was performed at later stage the redirect would mess up the flow.

Reply:
We could detect the proxy configuration and then we could do logout with a redirect, i.e. logout would happen after the dialog was closed.